packages/components/src/components/Table/Table.js

```diff
@@ -0,0 +1,173 @@
+                      button.handler(
+                        selectedRows,
+                        getBatchActionProps().onCancel
+                      );
+                    }}
+                  >
+                    {button.text}
+                  </TableBatchAction>
+                ))}
+              </TableBatchActions>
+              <TableToolbarContent>
+                {toolbarButtons.map(button => (
+                  <Button
+                    disabled={loading}
+                    onClick={button.handler}
+                    renderIcon={button.icon}
+                    key={`${button.text}Button`}
+                  >
+                    {button.text}
+                  </Button>
+                ))}
+              </TableToolbarContent>
+            </TableToolbar>
+            {loading ? (
+              <DataTableSkeleton
+                useZebraStyles
+                rowCount={1}
+                columnCount={headers.length}
+              />
+            ) : (
+              <CarbonTable>
+                <TableHead>
+                  <TableRow>
+                    {batchActionButtons.length > 0 && (
+                      <TableSelectAll {...getSelectionProps()} />
+                    )}
+                    {headers.map(header => {
+                      return (
+                        <TableHeader
+                          {...getHeaderProps({ header })}
+                          className="cellText"
+                          key={header.key}
+                        >
+                          {header.header}
+                        </TableHeader>
+                      );
+                    })}
+                  </TableRow>
+                </TableHead>
+                <TableBody>
+                  {rows.map(row => {
+                    return (
+                      <TableRow
+                        {...getRowProps({ row })}
+                        key={row.id}
+                        className="row"
+                      >
+                        {batchActionButtons.length > 0 && (
+                          <TableSelectRow {...getSelectionProps({ row })} />
+                        )}
+                        {row.cells.map(cell => (
+                          <TableCell
+                            key={cell.id}
+                            id={cell.id}
+                            className="cellText"
+                          >
+                            {cell.value}
+                          </TableCell>
+                        ))}
+                      </TableRow>
+                    );
+                  })}
+                </TableBody>
+              </CarbonTable>
+            )}
+          </TableContainer>
+        )}
+      />
+      {!loading &&
+        dataRows.length === 0 &&
+        selectedNamespace === ALL_NAMESPACES && (
+          <div className="noRows">
+            <p>{`No ${resource} created under any namespace.`}</p>
```

Comment:
Can we accept 2 additional props for the empty text? `emptyTextAllNamespaces` and `emptyTextSelectedNamespace` or something like that.

Also need to use react-intl here.

We can reduce duplication too as it's only the string that changes:

```js
{!loading &&
  rows.length === 0 && (
    <div className="noRows">
       {selectedNamespace === ALL_NAMESPACES ? emptyTextAllNamespaces : emptyTextSelectedNamespace}
    </div>
  )
}
```